Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -158,9 +160,11 @@ extension AWSClient {
                     httpMethod: httpMethod
                 )
                 return try self.createNioRequest(awsRequest)
-            }.whenSuccess { nioRequest in
-                _ = self.invoke(nioRequest)
-          }
+            }.then { nioRequest in
+                return self.invoke(nioRequest)
+            }.map { response in
```

Comment:
Why not return back the response instead of Void?

Reply:
hmm...

I thought about that. my reason not to was b/c the api spec says there is no output...

ex:

```
"AddTagsToCertificate":{
      "name":"AddTagsToCertificate",
      "http":{
        "method":"POST",
        "requestUri":"/"
      },
      "input":{"shape":"AddTagsToCertificateRequest"},
      "errors":[
        {"shape":"ResourceNotFoundException"},
        {"shape":"InvalidArnException"},
        {"shape":"InvalidTagException"},
        {"shape":"TooManyTagsException"}
      ]
    }
```

vs with an output

```
"DescribeCertificate":{
      "name":"DescribeCertificate",
      "http":{
        "method":"POST",
        "requestUri":"/"
      },
      "input":{"shape":"DescribeCertificateRequest"},
      "output":{"shape":"DescribeCertificateResponse"},
      "errors":[
        {"shape":"ResourceNotFoundException"},
        {"shape":"InvalidArnException"}
      ]
    }
```

So, if we return a NIO Response, are we conforming the the spec?

Reply:
oh and credit to @adam-fowler b/c he saw this and mentioned it in one of the other PRs :)

Reply:
@Yasumoto Surely, the only time you need the response is when there is an error. This can be dealt with in the same way as errors are dealt with for operations that return a non void response. See my comment above